noir/tooling/noir_js/src/program.ts

```diff
@@ -67,13 +67,13 @@ export class Noir {
    *
    * @example
    * ```typescript
-   * async generateFinalProof(input)
+   * async generateProof(input)
    * ```
    *
    */
-  async generateFinalProof(inputs: InputMap, foreignCallHandler?: ForeignCallHandler): Promise<ProofData> {
+  async generateProof(inputs: InputMap, foreignCallHandler?: ForeignCallHandler): Promise<ProofData> {
```

Comment:
I changed this to just `generateProof` as we no longer need the distinction between an intermediate and a final proof on the tooling layer. However, I know this is a breaking change so I could bring back `generateFinalProof` and `generateIntermediateProof` with a deprecation warning until we are ready to break. Or we can make the next release a breaking one for NoirJS users.

Reply:
I'm thinking we could just keep both `generateFinalProof` and `generateIntermediateProof` with a deprecation warning for a few weeks while people migrate to `generateProof`.

Reply:
For recursion, I think we should just make the breaking change as its not really an official feature yet

Reply:
@signorecello something to be aware of when this gets merged in

Reply:
Yeah I agree with @kevaundray we have some fierce users working with recursion but not enough to justify maintaining a deprecation working.

My opinion is that we can "practice" deprecation warnings because they will be needed at some point, but for now we're in alpha stage. Does it break? Go read docs. After a day working with `viem` that doesn't even have versioned docs, I really think we're way ahead of the standard tooling already.

`git push --force origin master`

Reply:
Thanks for the tag though, I'm happy to get rid of the difference between `intermediate proofs` and `final proofs`. Would appreciate if you could update the docs too. I'll take care of the examples once this gets merged

Reply:
@signorecello Would you prefer I update here as part of this PR or in a followup to this one you just made https://github.com/noir-lang/noir/pull/4187

Reply:
I made the docs PR (https://github.com/noir-lang/noir/pull/4199) as a followup to yours on the Noir repo